### ISSUE TITLE
Extern function symbols not recognized

### ISSUE THREAD

**Issue by Enkelmann** [BinaryAnalysisPlatform/bap#1131]
Recently BAP started to fail to recognize any function symbols of imported functions on some binaries. For example
```
bap /bin/echo --pass=dump-symbols | grep malloc
```
doesn't find anything, despite the binary importing the function `malloc`.

Tested for BAP 2.1 with LLVM 8.0. The binary in question is from a fresh Ubuntu 20.04 installation.

[echo.zip](https://github.com/BinaryAnalysisPlatform/bap/files/4785304/echo.zip)

**Comment by ivg** [comment]
@gitoleg, looping you in. @Enkelmann, did you install bap-objdump?

**Comment by Enkelmann** [comment]
Yes, bap-objdump is installed. Although I did not install it explicitly.

**Comment by ivg** [comment]
confirmed, reproducible, @gitoleg is working on it.

**Comment by ivg** [comment]
TL;DR; upgrade llvm to 9.0+.

It is not a bug neither it is an issue in BAP. The specificity of this binary is that all plt entries are having the form,
```
00000000000022d0 <getenv@plt>:
    22d0:       f3 0f 1e fa             endbr64 
    22d4:       f2 ff 25 a5 7b 00 00    bnd jmpq *0x7ba5(%rip)        # 9e80 <getenv@GLIBC_2.2.5>
    22db:       0f 1f 44 00 00          nopl   0x0(%rax,%rax,1)
```

Where `endbr64` is the new instruction introduced recently by Intel, which LLVM supports since 9.0 and BAP since 2.0 (#992).

**Comment by Enkelmann** [comment]
Upgrading to LLVM 9 helped, thanks! However, I still get the same problem on some other binaries. The example is again `/bin/echo`, this time taken from a Manjaro installation. Tested inside a docker container based on Ubuntu 18.04.

My local BAP 2.1 installation on Manjaro with LLVM 10 experiences the same problem, but with almost all binaries. But I don't know yet whether my local installation suffers from an unrelated problem.

[echo.zip](https://github.com/BinaryAnalysisPlatform/bap/files/4790969/echo.zip)

**Comment by ivg** [comment]
It is also to have an up-to-date objdump, e.g., 2.26 fails on such hardened binaries with
```

objdump -d echo 

echo:     file format elf64-x86-64

objdump: BFD (GNU Binutils for Ubuntu) 2.26.1 internal error, aborting at ../../bfd/elf64-x86-64.c:6089 in elf_x86_64_get_plt_sym_val
```

**Comment by ivg** [comment]
update, don't worry with the objdump versions

**Comment by ivg** [comment]
@Enkelmann, this is a completely different issue and indeed objdump is unable to identify symbols in those binaries, as well as `radare2`. But I think we can handle this. See #1135

**Comment by Enkelmann** [comment]
Thanks again, looking forward to a fix for #1135
